ocaml/sdk-gen/csharp/autogen/src/Session.cs

```diff
@@ -404,11 +405,12 @@ public void login_with_password(string username, string password, API_Version ve
         private void SetAPIVersion()
         {
             Dictionary<XenRef<Pool>, Pool> pools = Pool.get_all_records(this);
-            foreach (Pool pool in pools.Values)
+
+            if (pools.Values.Count > 0)
             {
+                var pool = pools.Values.First();
                 Host host = Host.get_record(this, pool.master);
                 APIVersion = Helper.GetAPIVersion(host.API_version_major, host.API_version_minor);

```

Comment:
Instead of getting the first, wouldn't it be better to get the maximum or the minimum version of all the pools?

Reply:
Theoretically speaking it would be more correct. Practically it does not make a difference since there is only one pool record in the pool.